project_euler/problem_112/sol1.py

```diff
@@ -0,0 +1,65 @@
+"""
+Problem 112: https://projecteuler.net/problem=112
+
+Working from left-to-right if no digit is exceeded by the digit to its left it is
+
+called an increasing number; for example, 134468.
+Similarly if no digit is exceeded by the digit to its right it is called a decreasing
+number; for example, 66420.
+We shall call a positive integer that is neither increasing nor decreasing a "bouncy"
+number, for example, 155349.
+Clearly there cannot be any bouncy numbers below one-hundred, but just over half of
+the numbers below one-thousand (525) are bouncy. In fact, the least number for which
+the proportion of bouncy numbers first reaches 50% is 538.
+Surprisingly, bouncy numbers become more and more common and by the time we reach
+21780 the proportion of bouncy numbers is equal to 90%.
+
+Find the least number for which the proportion of bouncy numbers is exactly 99%.
+"""
+
+
+def check_bouncy(n: int) -> bool:
+    """
+    Returns True if number is bouncy, False otherwise
+    >>> check_bouncy(6789)
+    False
+    >>> check_bouncy(-12345)
+    False
+    >>> check_bouncy(6.74)
+    False
+    >>> check_bouncy(132475)
+    True
+    >>> check_bouncy(-6548)
+    True
+    """
+    if not isinstance(n, int):
+        return False

```

Comment:
Shouldn't this be raising a `TypeError`?

Reply:
The function returns a `bool` value, so I thought it made more sense to simply return `False`

Reply:
`False` just signals that it is not bouncy but according to the rules, `6.74` is actually a bouncy number. So, if you're restricting the input to `int` then it makes sense to raise an error with the message saying the function only takes `int` as input.

Also, please test for lower numbers as well.

Reply:
Ok. I shall add more tests